src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -109,12 +108,27 @@ private async ValueTask<HttpResponseMessage> SendAsyncCore(HttpRequestMessage re
             DiagnosticListener diagnosticListener = s_diagnosticListener;
 
             Guid loggingRequestId = Guid.Empty;
-            Activity? activity = CreateActivity(request);
+            Activity? activity = StartActivity(request);
 
-            // Start activity anyway if it was created.
             if (activity is not null)
             {
-                activity.Start();
+                if (activity.IsAllDataRequested)
+                {
+                    // Add standard tags known before sending the request.
+                    KeyValuePair<string, object?> methodTag = DiagnosticsHelper.GetMethodTag(request.Method, out bool isUnknownMethod);
+                    activity.SetTag(methodTag.Key, methodTag.Value);
+                    if (isUnknownMethod)
+                    {
+                        activity.SetTag("http.request.method_original", request.Method.Method);
+                    }
+
+                    if (request.RequestUri is Uri requestUri && requestUri.IsAbsoluteUri)
+                    {
+                        activity.SetTag("server.address", requestUri.Host);
+                        activity.SetTag("server.port", requestUri.Port);
+                        activity.SetTag("url.full", DiagnosticsHelper.GetRedactedUriString(requestUri));
```

Comment:
Is the idea behind having all such tags to make it easier to query for info later on?
Otherwise, this is all redundant info (address, port, scheme).

Reply:
I assume yes. I wonder if the redundancy issue was ever raised in semconv discussions. @lmolkova any insights?

Reply:
yes, `server.address` and `server.port` are there to allow querying later without parsing the URL. Also some apps/backends calculate metrics-over-traces and it's important to have host and port readily available without requiring to parse URL.

It's a trade-off and redundancy is intentional,